### PR TITLE
Upgrade go-waku to v0.8.0

### PR DESCRIPTION
**Pull request by snormore**
Upgrades go-waku from v0.7.0 to v0.8.0.

Release notes: https://github.com/waku-org/go-waku/releases/tag/v0.8.0

### CODE REVIEW

File: dev/docker/Dockerfile [hunk 1]
```diff
@@ -10,7 +10,7 @@ COPY . .
 
 # Build the final node binary
 ARG GIT_COMMIT=unknown
-RUN go build -ldflags="-X 'main.Commit=$GIT_COMMIT'" -o bin/xmtpd cmd/xmtpd/main.go
+RUN go build -tags="gowaku_no_rln" -ldflags="-X 'main.Commit=$GIT_COMMIT'" -o bin/xmtpd cmd/xmtpd/main.go
```
Comment (snormore) [inline dev/docker/Dockerfile:13]: The docker build was failing with this:
```
44.72 # command-line-arguments
44.72 /usr/local/go/pkg/tool/linux_arm64/link: running gcc failed: exit status 1
44.72 /usr/lib/gcc/aarch64-alpine-linux-musl/12.2.1/../../../../aarch64-alpine-linux-musl/bin/ld: /go/pkg/mod/github.com/waku-org/go-zerokit-rln-arm@v0.0.0-20230914230828-f14923ed4dac/rln/../libs/aarch64-unknown-linux-gnu/librln.a(std-d6de9543d35dcb36.std.0c288059-cgu.0.rcgu.o): in function `std::sys::unix::os::glibc_version':
44.72 /rustc/69f9c33d71c871fc16ac445211281c6e7a340943/library/std/src/sys/unix/os.rs:660: undefined reference to `gnu_get_libc_version'
44.72 /usr/lib/gcc/aarch64-alpine-linux-musl/12.2.1/../../../../aarch64-alpine-linux-musl/bin/ld: /go/pkg/mod/github.com/waku-org/go-zerokit-rln-arm@v0.0.0-20230914230828-f14923ed4dac/rln/../libs/aarch64-unknown-linux-gnu/librln.a(std-d6de9543d35dcb36.std.0c288059-cgu.0.rcgu.o): in function `std::sys::unix::net::on_resolver_failure':
44.72 /rustc/69f9c33d71c871fc16ac445211281c6e7a340943/library/std/src/sys/unix/net.rs:528: undefined reference to `__res_init'
44.72 collect2: error: ld returned 1 exit status
```

Noticed a flag to disable rln compilation, so it was the easy solution until it's needed

**Comment by snormore** [comment]
I'll deploy this to dev for a bit first, it's still a pretty new release (1 week old).

File: pkg/server/server.go [hunk 2]
```diff
@@ -85,9 +86,10 @@ func New(ctx context.Context, log *zap.Logger, options Options) (*Server, error)
 
 	s.ctx, s.cancel = context.WithCancel(ctx)
 
+	promReg := prometheus.NewRegistry()
```
Comment (snormore) [inline pkg/server/server.go:89]: Had to do a bit of work with metrics to consolidate into the same prometheus registry or else we're losing the api metrics.

Comment (snormore) [reply pkg/server/server.go:89]: We'll have to update the DD references to use the `xmtpd.` prefix on metrics instead of `waku.` prefix after merging this

**Review by neekolas — APPROVED** [review]
LGTM

**Pull request merged**